dbt/include/global_project/macros/adapters/snowflake.sql

```diff
@@ -1,10 +1,28 @@
+{% macro clusterby(_clusterby) %}
+  {%- if _clusterby is not none -%}
+    cluster by (
+      {%- if _clusterby is string -%}
+        {%- set _clusterby = [_clusterby] -%}
+      {%- endif -%}
+      {%- for item in _clusterby -%}
+        "{{ item }}"
```

Comment:
We're currently experiencing a ton of pain around quoting on Snowflake. I think our default approach going forward will be to _not_ quote identifiers in dbt, but instead ask the user to explicitly quote fields like this if desired. Do you buy that?

Reply:
Another benefit of not quoting:

You made a great example of using `cluster_by` in the `sample.dbt_project.yml` file below:
```
                clusterby: ["date_day", "ad_group_id % 10"]
```

If dbt adds quotes, then I think the expression `ad_group_id % 10` will look like an identifier, and Snowflake will do the wrong thing

Reply:
Yes, I buy that! We are definitely in the "don't quote identifiers and let the database do it's thing" camp. Allowing users to override that behavior on a case-by-case basis makes sense.


We have an edge case in our Vertica to Snowflake migration, where we manually created a handful of views that quote identifiers lower-cased versions of the table names. This is so that our pre-existing Tableau reports, which quote a lower-cased version of the table name, continue to function without change.